### PR TITLE
[WIP] add router disposal to close method

### PR DESCRIPTION
**Pull request by 5ebastianMeier**
Closes https://github.com/essential-projects/http_node/issues/4

## What did you change?

* Implemented the `dispose` method on the `HttpExtension`

## How can others test the changes?

See https://github.com/process-engine/process_engine_meta/pull/72

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/http_extension.ts [hunk 1]
```diff
@@ -139,12 +139,17 @@ export class HttpExtension implements IHttpExtension {
...
-  public close(): void {
+  public async close(): Promise<void> {
 
     if (this.server) {
 
       this.server.close();
     }
+
+    for (const routerName in this.routers) {
+      const router: any = this.routers[routerName];
```
Comment (cmg-dev) [inline src/http_extension.ts]: _ missing in access to `this.routers`...

File: src/http_extension.ts [hunk 2]
```diff
@@ -139,12 +139,17 @@ export class HttpExtension implements IHttpExtension {
...
 
-  public close(): void {
+  public async close(): Promise<void> {
 
     if (this.server) {
 
       this.server.close();
     }
+
+    for (const routerName in this.routers) {
```
Comment (cmg-dev) [inline src/http_extension.ts]: _ missing.

Comment (cmg-dev) [reply src/http_extension.ts]: But this Constructs looks awkward anyway...

```ts
for (const routerName in this._routers) {
   const router: any = this._routers[routerName];
(...)
```

is this correct? 

for each over the _routers list (which is btw of type `any`:( ) should give you the same thing...

Or am I totally misstaken about the type in this._routers?

**Comment by ElRaptorus** [comment]
This is going to be included in the next major release.

**Pull request closed without merging**